cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -136,29 +142,24 @@ public void destroy() {
     // Nothing to do
   }
 
-  private Properties generateKafkaConfig(int brokerId, String zkConnectStr, String hostname, int port,
-                                         int numPartitions, int replicationFactor, String logDir) {
-    Preconditions.checkState(port > 0, "Port number is invalid.");
-    Preconditions.checkState(numPartitions > 0, "Num partitions should be greater than zero.");
+  private int tryParse(String val) {
```

Comment:
`parseInt` will throw NumberFormatException if its not a string of a number; it won't return null.

The following is what was throwing an exception, because Integer.parseInt( null ) was called on line 85, because a default value for `NUM_PARTITIONS_CONFIG` was not being defined:
https://github.com/caskdata/cdap/pull/5888/files#diff-3fde3c97f86628a648cb1a78420b0abdR85